### ISSUE TITLE
Mod panel authentication.

### ISSUE THREAD

**Issue by Semisol** [micahlt/modchat#74]
Like fr, you always seem to get it wrong.
You need to authenticate the post endpoints and stuff, like I can just `mitmproxy` the panel and use it... And `Referrer` + `Origin` can be forged. Reported this in e-mail to keep it private, no proper fixes, PLEASE NOW FIX IT.

Your security is so flawed I am about to die ._.

**Comment by CodeLikeCrazE** [comment]
You do realize that sending your reply using email doesn't make it any more private, right?

________________________________
From: Semisol ***@***.***>
Sent: Friday, April 2, 2021 8:05 PM
To: micahlt/modchat ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [micahlt/modchat] Mod panel authentication. (#74)


Like fr, you always seem to get it wrong.
You need to authenticate the post endpoints and stuff, like I can just mitmproxy the panel and use it... And Referrer + Origin can be forged. Reported this in e-mail to keep it private, no proper fixes, PLEASE NOW FIX IT.

Your security is so flawed I am about to die ._.

**Comment by micahlt** [comment]
@CodeLikeCrazE Yeah, we know.  He's referring to the fact that he emailed me (hi@micahlindley.com) directly.  

@Semisol I'm no security expert, and I'm a frontend dev.  Security is not my forte.  The current codebase is a mess, so I'm fully rebuilding Modchat from the ground up.  If you'd like to assist me, please visit the [micahlt/modchat-server](https://github.com/micahlt/modchat-server) repo.

**Comment by Semisol** [comment]
> @CodeLikeCrazE Yeah, we know. He's referring to the fact that he emailed me ([hi@micahlindley.com](mailto:hi@micahlindley.com)) directly.
> 
> @Semisol I'm no security expert, and I'm a frontend dev. Security is not my forte. The current codebase is a mess, so I'm fully rebuilding Modchat from the ground up. If you'd like to assist me, please visit the [micahlt/modchat-server](https://github.com/micahlt/modchat-server) repo.

I would. But ENDPOINTS need authentication, not just the HTML.
